test/steps/v2_steps.py

```diff
@@ -2029,15 +2029,59 @@ def wait_for_app_txn_confirm(context):
 @given("I remember the new application ID.")
 def remember_app_id(context):
     if hasattr(context, "acl"):
-        context.current_application_id = context.acl.pending_transaction_info(
-            context.app_txid
-        )["txresults"]["createdapp"]
+        app_id = context.acl.pending_transaction_info(context.app_txid)[
+            "txresults"
+        ]["createdapp"]
     else:
-        context.current_application_id = (
-            context.app_acl.pending_transaction_info(context.app_txid)[
-                "application-index"
-            ]
-        )
+        app_id = context.app_acl.pending_transaction_info(context.app_txid)[
+            "application-index"
+        ]
+
+    context.current_application_id = app_id
+    if not hasattr(context, "app_ids"):
+        context.app_ids = []
+
+    context.app_ids.append(app_id)
+
+
+@then(
+    "I get the account address for the current application and see that it matches the app id's hash"
+)
+def assert_app_account_is_the_hash(context):
+    app_id = context.current_application_id
+    expected = encoding.encode_address(
+        encoding.checksum(b"appID" + app_id.to_bytes(8, "big"))
+    )
+    actual = logic.get_application_address(app_id)
+    assert (
+        expected == actual
+    ), f"account-address: expected [{expected}], but got [{actual}]"
```

Comment:
This test is not very strong because the expected value is being calculated with essentially the same implementation as the actual value. But I suppose it's better than no test, so it's fine to keep it

Reply:
Yep. It was stronger before I realized that we already had the functionality 🤦 . 

But like you said, some test is better than no test.